libopae/plugins/xfpga/buffer.c

```diff
@@ -229,15 +231,7 @@ fpga_result __FPGA_API__ xfpga_fpgaPrepareBuffer(fpga_handle handle, uint64_t le
 		}
 	}
 
-	/* Set ioctl fpga_port_dma_map struct parameters */
-	struct fpga_port_dma_map dma_map = {.argsz = sizeof(dma_map),
-					    .flags = 0,
-					    .user_addr = (__u64) addr,
-					    .length = (__u64) len,
-					    .iova = 0};
-
-	/* Dispatch ioctl command */
-	if (ioctl(_handle->fddev, FPGA_PORT_DMA_MAP, &dma_map) != 0) {
+	if (opae_port_map(_handle->fddev, addr, len, &io_addr)) {
```

Comment:
Don't use wrapper for IOCTL ,  too many wrappers causes code unmaintainable

store driver version enum fpga_token  
enum {
fpga_driver_devp_lateset_version,
fpga_driver_ups_version,
}
if( token -> drv_ver == fpga_driver_devp_lateset_version )
{ 
  ioctl()

}else  {
 ioctl()
}

Reply:
@anandaravuri I see it the other way around.  If the code is sprinkled with this sort of pattern..
```c
if (<this version>) {
  do_this();
} else {
  do_that();
}
```
Then the maintenance headache is increased, especially for handling new versions.  With the abstraction, the code that needs to change can be co-located, which reduces the maintenance overhead.

Reply:
Yes, I agree with @tswhison . Putting ioctl calls in one file greatly increases maintainability and readability. The approach you're proposing, @anandaravuri , tends to increase function size and code duplication.